pkg/kubelet/dockertools/exec.go

```diff
@@ -121,6 +126,14 @@ func (*NativeExecHandler) ExecInContainer(client DockerInterface, container *doc
 	if err != nil {
 		return err
 	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
+			// FIXME: we should kill the process in the container,
+			// but I couldn't find anything in the Docker API docs
+			// on how to do it with the *Exec APIs.
```

Comment:
I need to check, wondering if the call to `client.StartExec` is blocking or not until the command terminates, given that we set `Detach: false`.
If it blocks till the command terminates, then I don't understand the `for` loop below.